src/main/java/com/manardenza/entity/Reservation.java

```diff
@@ -7,9 +7,31 @@
 
 @Data
 @AllArgsConstructor
-public class Reservation {
+public class Reservation extends AbstractObject {
 
     private Date reservedFrom;
     private Date reservedTo;
     private User reservedUser;
+    private Room reservedRoom;
+    private Hotel reservedHotel;
+
+    public boolean newReservedOk(Date newReservedFrom, Date newReservedTo) {
+        if (newReservedFrom.after(reservedFrom) && newReservedFrom.before(reservedTo)) {
+            return false;
+        } else if (newReservedTo.after(reservedFrom) && newReservedTo.before(reservedTo)) {
+            return false;
+        } else if (newReservedFrom.before(reservedFrom) && newReservedTo.after(reservedTo)) {
+            return false;
+        } else if (newReservedFrom.equals(reservedFrom)) {
+            return false;
+        } else if (newReservedFrom.equals(reservedTo)) {
+            return false;
+        } else if (newReservedTo.equals(reservedFrom)) {
+            return false;
+        } else if (newReservedTo.equals(reservedTo)) {
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png 'Severity: MINOR') This branch's code block is the same as the block for the branch on line 29. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AS1871)
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png 'Severity: MINOR') Replace this if-then-else statement by a single return statement. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AS1126)